chaosk8s/pod/actions.py

```diff
@@ -46,21 +47,111 @@ def terminate_pods(label_selector: str = None, name_pattern: str = None,
     """
+
+    api = create_k8s_api_client(secrets)
+    v1 = client.CoreV1Api(api)
+
+    pods = _select_pods(v1, label_selector, name_pattern,
+                        all, rand, mode, qty, ns, order)
+
+    body = client.V1DeleteOptions()
+    if grace_period >= 0:
+        body = client.V1DeleteOptions(grace_period_seconds=grace_period)
+
+    for p in pods:
+        v1.delete_namespaced_pod(p.metadata.name, ns, body=body)
+
+
+def exec_in_pods(label_selector: str = None, name_pattern: str = None,
+                 all: bool = False, rand: bool = False,
+                 mode: str = "fixed", qty: int = 1,
+                 ns: str = "default", order: str = "alphabetic",
+                 secrets: Secrets = None, cmdstr: str = None,
+                 container_name: str = None):
+    """
+    Exceute a commnand in the specified pod's container.
+    Select the appropriate pods by label and/or name patterns.
+    Whenever a pattern is provided for the name, all pods retrieved will be
+    filtered out if their name do not match the given pattern.
+
+    If neither `label_selector` nor `name_pattern` are provided, all pods
+    in the namespace will be selected for termination.
+
+    If `all` is set to `True`, all matching pods will be affected.
+
+    Value of `qty` varies based on `mode`.
+    If `mode` is set to `fixed`, then `qty` refers to number of pods affected.
+    If `mode` is set to `percentage`, then `qty` refers to
+    percentage of pods, from 1 to 100, to be affected.
+    Default `mode` is `fixed` and default `qty` is `1`.
+
+    If `order` is set to `oldest`, the retrieved pods will be ordered
+    by the pods creation_timestamp, with the oldest pod first in list.
+
+    If `rand` is set to `True`, n random pods will be affected
+    Otherwise, the first retrieved n pods will be used
+    """
+
+    api = create_k8s_api_client(secrets)
+    v1 = client.CoreV1Api(api)
+
+    pods = _select_pods(v1, label_selector, name_pattern,
+                        all, rand, mode, qty, ns, order)
+
+    if cmdstr:
+        exec_command = cmdstr.split()
+    else:
+        logger.debug("No command specified to be executed. skip activity")
+        return
+
+    for po in pods:
+        logger.debug("Picked pods '{p}' for command execution {c}".format(
+            p=po.metadata.name, c=exec_command))
+        if not any(c.name == container_name for c in po.spec.containers):
+            logger.debug("Pod {p} do not have container named '{n}'".format(
+                p=po.metadata.name, n=container_name))
+            continue
+
+        stream.stream(v1.connect_get_namespaced_pod_exec,
+                      po.metadata.name,
+                      ns,
+                      container=container_name,
+                      command=exec_command,
+                      stderr=True,
+                      stdin=False,
+                      stdout=True,
+                      tty=False)
+
+
+def _sort_by_pod_creation_timestamp(pod):
```

Comment:
could you add typing annotation (args and return value)?

Reply:
Well, this is just an helper function to sort pods by their create timestamp. 
I have added a little comment about it in the function

Reply:
Sure, but it's good convention to set the annotation nonetheles :)

Something like:

```
def _sort_by_pod_creation_timestamp(pod: client.V1Pod) -> int:
```